### ISSUE TITLE
[Bug] 

### ISSUE THREAD

**Issue by reinrl** [pqina/filepond#776]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Have you updated FilePond and its plugins?

- [X] I have updated FilePond and its plugins

### Describe the bug

I know that FilePond advertises itself as accessible, and I see some aria attributes on the underlying that is created...but I can't see quite how to add any additional a11y-related props (e.g., aria-invalid), nor how to control what is added automatically (e.g., having aria-labelledby reference any label or error information that might be a part of my implementation of a file upload field).

I have tried passing additional props to the existing React adapter, as well as making a copy of the existing React adapter locally in my project and trying to set additional properties on the rendered input (which I assume means that somewhere in the bowels of the more generic FilePond project, it really just uses that input field as a reference, and then creates its own instead?).

I would love to learn that I am just missing something existing in the API (I created a [question on StackOverflow](https://stackoverflow.com/questions/69916932/filepond-a11y-questions) as well) - but alternately I have filed this issue on the project github in case this is just an issue with the existing implementation (or possibly even morphs into a feature request).

### Reproduction

Example simply trying to put an additional attribute (aria-invalid) on the input that FilePond is attached to: https://codesandbox.io/s/new-water-y0p96?file=/src/index.js

### Environment

```markdown
- Device:
- OS:
- Broser:
```

**Comment by rikschennink** [comment]
Hi, Yes all internals are managed by FilePond and there's currently no way to dynamically add custom attributes to internal fields (apart from id / class).

On which internal elements would you like to add attributes?

**Comment by reinrl** [comment]
My current need stems directly from an A11Y standpoint - being able to have an `aria-invalid` prop on the underlying input field (with the value accurately reflecting the current state of the field in general), and having the `aria-labelledby` prop be something that is able to be hooked into (today is points to the label within the upload zone itself - which is marked with aria-hidden?? - while my implementation may or may not also have a label above the upload zone and conditionally a display an error message below the upload zone, both of which would be considered as contributing to the labeling of the upload zone.

**Comment by reinrl** [comment]
@rikschennink I noticed that you changed the title - does that mean that this would be a feature that you are considering?

**Comment by rikschennink** [comment]
@reinrl Definitely, I want FilePond to be accessible and it's indeed a bit weird that when the label hides there is no other label describing the field.

However I'm super busy so any PR requests are welcome. 

Additionally, I'm looking at developing version 5 of FilePond and gathering requirements.

**Comment by reinrl** [comment]
@rikschennink So, I think that I have hit a bit of a roadblock (and need some further guidance from a philosophical perspective from the project owner).

So I started with the `aria-invalid` prop as my first change. I went into `/src/js/app/view/browser.js`, found the piece where the other props for the `<input>` were being set, and adding a line like the following:

```js
// element needs to have a way to state its current validity
attr(
  root.element,
  'aria-invalid',
  // required but no files yet, or...
  (!!root.query('GET_REQUIRED') && root.query('GET_STATUS') === Status.EMPTY) ||
    // ...one or more files are in error state
    root.query('GET_STATUS') === Status.ERROR
);
```

The thinking being that the field should probably consider itself invalid whenever either it is in an error status, or when it is required and no files have yet been uploaded.

I began to add unit test coverage, and ran into a few different things:
- The underlying input field doesn't seem to be rendered at the same time as everything else. The only way that I could actually "see" the input field was to make the test case asynchronous, and explicitly poll until an input tag became available (to see that the initial state of the prop reflected that the field was initially valid, as an example). Is it intentional that the rendering of the input is somehow delayed? Or is there perhaps something to do with not waiting for a promise resolution somewhere in the bowels of FilePond that I missed upon initial review?
- For my next test case, I wanted to trigger an error to confirm that the input property gets updated as expected. I used an example from `addFile.test.js` as a starting point to manually create a file and then "upload" it via a call to `instance.addFile()`. I set the instance configuration to make the max file size "0", which was more than enough to force an error when I was playing around on [this codepen](https://codepen.io/reinrl/pen/KKvLVjj)...until I realized that file size validation was an additional plugin (which sparked a whole other topic for me - why is max file size a core property that can be passed in, but the actual use of that property requires loading a plugin?). Not wanting to use a plugin in a core unit test, I decided to move on and find a different way to force an error - but `callbacks.test,js` didn't give me any starting points, as there isn't an existing test case for the `onerror` callback. Any hints as to the best way to programmatically force an error for purposes of unit testing?

**Comment by rikschennink** [comment]
@reinrl 

1. I don't know. But what I do know is that aria attributes should be used when HTML attributes don't cut it. So would a `required` attribute not catch this? See also the `checkValidity` property. I'm sorry I'm not more helpful on this, it has been a while since I worked on the UI code so it's not front of mind.

2. There's a lot going on internally to make sure animations are smooth and not blocked by DOM interaction, that is most likely why the field isn't there immediately. Maybe you can use a `MutationObserver` to catch the addition of the field.

3. From the top of my head I don't think there are validations in the base library to trigger an error but you can set an incorrect server URL and it'll throw when it fails to upload.

```js
FIlePond.create({
    server: './invalid-path'
})
```

**Comment by reinrl** [comment]
@rikschennink Good call on the server error - as a starting point, I have created pull request #779 to add test coverage for `onerror`.
